### ISSUE TITLE
How to handle GMSH mesh format version 4?

### ISSUE THREAD

**Issue by jbrezmorf** [flow123d/flow123d#1220]
1. Flow123d only supports GMSH ASCII mesh format version 2. Adding support for the version 4 is planed, but that will take some time.

2. Preferably generate your meshes directly to the MSH2 format, e.g.:
```
    gmsh -2 -format msh2 your_file.geo
```

3. Having the version 4 file ou can convert the file through GMSH GUI:
    1. open your version 4 file
    2. File > Export
    3. Select `Mesh - Gmsh MSH (*.msh)`
    4. Select `Version 2 ASCII`, DO NOT CHECK `Save all elements.`

4.  From the command line:
```
    gmsh your_msh4.msh -save -format msh2 -o your_msh2..msh
```

**Comment by jbrezmorf** [comment]
GMSH 4 format reader / writer. 
Proposed as a library with Python interface since we need it in Python as well.
Current gmsh_io.py is a buggy mess. 
We can put it into BGEM that way we force BGEM to be part of the installation image.

**Comment by jbrezmorf** [comment]
The biggest issue is a suitable interface.

**Comment by jbrezmorf** [comment]
Proposed to use GMSH library and its C++ API for both reader and the writer.

**Comment by jbrezmorf** [comment]
Will be solved within #1437.
